internal/bminventory/inventory.go

```diff
@@ -1660,6 +1667,28 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func alignClusterPlatformParams(params installer.V2UpdateClusterParams, cluster *common.Cluster) error {
+	paramsUserManagedNetworking := params.ClusterUpdateParams.UserManagedNetworking
+
+	// Attempts to set UserManagedNetworking=false and Platform=none or UserManagedNetworking=true and Platform=baremetal
+	// will result StatusBadRequest error
+	if paramsUserManagedNetworking != nil && params.ClusterUpdateParams.Platform != nil {
+		platformType := *params.ClusterUpdateParams.Platform.Type
+		if !*paramsUserManagedNetworking && platformType == models.PlatformTypeNone || *paramsUserManagedNetworking && models.PlatformTypeBaremetal == platformType {
+			return common.NewApiError(http.StatusBadRequest, errors.Errorf("Can't set %s platform with user-managed-networking %v", platformType, *paramsUserManagedNetworking))
+		}
+	}
+
+	if swag.BoolValue(paramsUserManagedNetworking) && *cluster.Platform.Type == models.PlatformTypeBaremetal {
+		params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
+	} else if params.ClusterUpdateParams.Platform != nil && *params.ClusterUpdateParams.Platform.Type == models.PlatformTypeNone {
```

Comment:
So if platform == none it will just override params.ClusterUpdateParams.UserManagedNetworking to true?
why?

Reply:
It will set the platform_type to none and will update UserManagedNetworking=true.
The update will be later when UserManagedNetworking and platform is updated